src/core/server/saved_objects/migrations/core/build_active_mappings.test.ts

```diff
@@ -183,6 +183,7 @@ describe('diffMappings', () => {
       _meta: {
         migrationMappingPropertyHashes: { foo: 'bar' },
       },
+      // @ts-expect-error
```

Comment:
What's the error here?

Reply:
Error msg is 
```
(property) IndexMapping.dynamic?: boolean | "strict" | undefined
Type '"abcde"' is not assignable to type 'boolean | "strict" | undefined'.ts(2322)
types.ts(136, 3): The expected type comes from property 'dynamic' which is declared here on type 'IndexMapping'
```

There are two ways to change:
1) add description:
pros: don't need to change the test cases
cons: to explain clearly, descriptions might be long. for example,
```
Type '"abcde"' is not assignable to type 'boolean | "strict" | undefined'.ts. This is expected because we are assigning a different value to dynamic to testify diffMapping function.
```
2) change the original test to `dynamic: true`
pros: code is clean
cons: change the test case. the original test case also shows a more realistic case that user might assign any values by mistake

I prefer the first way. Any suggestions? @tmarkley

Reply:
will add some explain here